### ISSUE TITLE
KubeDB operator should recover inconsistent Postgres replicas

### ISSUE THREAD

**Issue by stephan2012** [kubedb/project#682]
After a system crash, a KubeDB operated Postgres 11.2 database replica fails to start:

```
I1111 12:49:20.732359       1 log.go:172] FLAG: --alsologtostderr="false"
I1111 12:49:20.732397       1 log.go:172] FLAG: --bypass-validating-webhook-xray="false"
I1111 12:49:20.732402       1 log.go:172] FLAG: --enable-analytics="false"
I1111 12:49:20.732410       1 log.go:172] FLAG: --help="false"
I1111 12:49:20.732415       1 log.go:172] FLAG: --log-flush-frequency="5s"
I1111 12:49:20.732421       1 log.go:172] FLAG: --log_backtrace_at=":0"
I1111 12:49:20.732439       1 log.go:172] FLAG: --log_dir=""
I1111 12:49:20.732444       1 log.go:172] FLAG: --logtostderr="false"
I1111 12:49:20.732449       1 log.go:172] FLAG: --stderrthreshold="0"
I1111 12:49:20.732453       1 log.go:172] FLAG: --use-kubeapiserver-fqdn-for-aks="true"
I1111 12:49:20.732458       1 log.go:172] FLAG: --v="3"
I1111 12:49:20.732462       1 log.go:172] FLAG: --vmodule=""
We want "ha-asa-alarm-b-db-1" as our leader
I1111 12:49:20.770284       1 leaderelection.go:205] attempting to acquire leader lease  default/ha-asa-alarm-b-db-leader-lock...
Running as Replica
Attempting pg_isready on primary
Attempting query on primary
2019-11-11 12:49:21.564 GMT [17] LOG:  skipping missing configuration file "/etc/config/user.conf"
2019-11-11 12:49:21.564 GMT [17] LOG:  skipping missing configuration file "/etc/config/user.conf"
2019-11-11 12:49:21.572 UTC [17] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2019-11-11 12:49:21.572 UTC [17] LOG:  listening on IPv6 address "::", port 5432
2019-11-11 12:49:21.577 UTC [17] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2019-11-11 12:49:21.599 UTC [29] LOG:  database system was interrupted; last known up at 2019-11-11 12:49:20 UTC
2019-11-11 12:49:21.689 UTC [29] LOG:  entering standby mode
2019-11-11 12:49:21.689 UTC [29] FATAL:  requested timeline 3 is not a child of this server's history
2019-11-11 12:49:21.689 UTC [29] DETAIL:  Latest checkpoint is at 1/B6000060 on timeline 2, but in the history of the requested timeline, the server forked off from that timeline at 0/1D000000.
2019-11-11 12:49:21.690 UTC [17] LOG:  startup process (PID 29) exited with exit code 1
2019-11-11 12:49:21.690 UTC [17] LOG:  aborting startup due to startup process failure
2019-11-11 12:49:21.692 UTC [17] LOG:  database system is shut down
I1111 12:49:21.693114       1 log.go:172] exit status 1
```

Symptoms look similar to #677.  However, reducing the number of replicas to 1 and deleting the PVC does not help. This raises two questions:

1. How to make KubeDB recover this situation?
2. Should KubeDB handle this situation automatically?

**Comment by stephan2012** [comment]
Nobody?

**Comment by Hugome** [comment]
I have the same issue (In 2 different clusters), did you find a solution ? (Like you, reducing the number of replicas to 1 and deleting PVC, didn't change a thing)

**Comment by tamalsaha** [comment]
This issue has been resolved in KubeDB release v2021.03.17 . Please open a new bug if you still see this issue.
